internal/controller/nim/account_controller.go

```diff
@@ -144,9 +145,11 @@ func (r *AccountReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		},
 	}
 
-	defer func() {
-		r.updateStatus(ctx, req.NamespacedName, *targetStatus)
-	}()
+	accountSubject := types.NamespacedName{Name: req.Name, Namespace: req.Namespace}
+
+	if updateErr := utils.UpdateStatus(ctx, accountSubject, *targetStatus, r.Client); updateErr != nil {
```

Comment:
are these `updateStatus` added on too many different places really needed?

Reply:
@spolti the use case is we want to post the status any time there is interaction witht he NIM API to give real time feedback to the user. If there is another way to accomplish this please advise otherwise I would say this achieves our goal

Reply:
@spolti did you have a chance to take another look at my response

Reply:
I see, I am still not sure if it is needed, I mean, a reconcile loop shouldn't take too long to finish and with the status being updated so many times would be even tough to see everything that is being updated (unless I am missing details on how these updates will be displayed to the user).

One solution would be to put this status into a map and call the statusUpdate function only once a the end of the reconcile loop.

Reply:
@spolti I don't think that would be a valid solution. The use case here is we make dozens of calls to the NVIDIA API for each model currently the list is at 67. Meaning each reconciliation is expected to send roughly 71 consecutive requests. The objective here is get real time feed back on if and when any of those calls fail.

Reply:
@spolti when you have time please see my comment above and advise. We were wanting to get this in for 2.19